lib/active_model/serializer/field.rb

```diff
@@ -27,16 +27,29 @@ def value(serializer)
       def excluded?(serializer)
         case condition_type
         when :if
-          !serializer.public_send(condition)
+          !condition_lambda.call(serializer)
         when :unless
-          serializer.public_send(condition)
+          condition_lambda.call(serializer)
         else
           false
         end
       end
 
       private
 
+      def condition_lambda
+        case condition
+        when Symbol
+          -> (serializer) { serializer.public_send(condition) }
```

Comment:
I'm not sure building a throwaway lambda here is the best way. How about the following
```ruby
def condition_lambda(serializer)
  case condition
  when Symbol
    serializer.public_send(condition)
  when String
    serializer.instance_eval(condition)
  when Proc
    condition.call(serializer)
...
```